### ISSUE TITLE
Handling multiple concept fields/planes/domains

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#621]
As discussed with @opoudjis , this work integrates a few components we already have.

Use case: I want to have a concept dictionary that is composed of multiple concept fields, represented in Metanorma. This need is already shown in ISO 10303-2.

ISO 10303-2 contains:
* normal Clause 3 terms in clause 3
* EXPRESS application integration modules in clause 4
* EXPRESS module-integration modules in clause 5

Example term from clause 3 (general ISO concept):
![image](https://user-images.githubusercontent.com/11865/163253658-d8f40179-e8f2-4cc7-b8c4-582b7eee34e7.png)

Example term from clause 4 (EXPRESS concept): 
![image](https://user-images.githubusercontent.com/11865/163255074-ba76c82d-391d-4ff1-a3fe-6207a89939b2.png)

These 3 types of concepts technically belong to different "concept fields" / "concept planes" / "concept domains". And the individual concepts can be related, e.g. a concept in clause 4 can be the implementation of an English concept in clause 3.

The representations of a Clause 3 term and an EXPRESS entity concept would also likely be different, as the data structures are different.

Notice that for ISO terms, we explicitly or implicitly define:
* Data structure: ISO 10241-1, etc.
* Input syntax: as in the concept AsciiDoc input syntax used by Metanorma
* Presentation: as defined in ISO 10241-1 / ISO DIR 2 and implemented in metanorma-iso

Basically, we want to achieve this:
* Allow the author to define a data structure using a data specification language (a schema)
* Allow the author to define the input syntax (based on the data specification)
* Allow the author to specify a presentation template, e.g. as in Annotated EXPRESS, a Liquid-enabled Metanorma AsciIDoc template.

In order to extend this functionality to other concept fields/planes/domains, we can:
* For the data specification language, we use LutaML
* For the input syntax, define using LutaML
* For the presentation, use the LutaML output block

Then in order to render non-ISO concepts, it would be simply to enumerate them from some external YAML file.

**Comment by opoudjis** [comment]
@ronaldtse I have no idea what you are asking for, and absolutely none of this:

````
Allow the author to define a data structure using a data specification language (a schema)
Allow the author to define the input syntax (based on the data specification)
Allow the author to specify a presentation template, e.g. as in Annotated EXPRESS, a Liquid-enabled Metanorma AsciIDoc template.
````

is any responsibility of mine at all. Neither is Lutaml. I am unassigning myself this ticket.
